src/hardware/serialport/misc_util.cpp

```diff
@@ -135,83 +133,87 @@ TCPClientSocket::~TCPClientSocket()
 	if(listensocketset) SDLNet_FreeSocketSet(listensocketset);
 }
 
-bool TCPClientSocket::GetRemoteAddressString(Bit8u* buffer) {
-	IPaddress* remote_ip;
-	Bit8u b1, b2, b3, b4;
+bool TCPClientSocket::GetRemoteAddressString(uint8_t *buffer)
+{
+	IPaddress *remote_ip;
+	uint8_t b1, b2, b3, b4;
 	remote_ip=SDLNet_TCP_GetPeerAddress(mysock);
 	if(!remote_ip) return false;
 	b4=remote_ip->host>>24;
 	b3=(remote_ip->host>>16)&0xff;
 	b2=(remote_ip->host>>8)&0xff;
 	b1=remote_ip->host&0xff;
-	sprintf((char*)buffer,"%u.%u.%u.%u",b1,b2,b3,b4);
+	sprintf((char*)buffer,"%" PRIu8 ".%" PRIu8 ".%" PRIu8 ".%" PRIu8,
+	        b1, b2, b3, b4);
 	return true;
 }
 
-bool TCPClientSocket::ReceiveArray(Bit8u* data, Bitu* size) {
-	if(SDLNet_CheckSockets(listensocketset,0))
-	{
-		Bits retval = SDLNet_TCP_Recv(mysock, data, static_cast<int>(*size));
-		if(retval<1) {
-			isopen=false;
-			*size = 0;
+bool TCPClientSocket::ReceiveArray(uint8_t *data, size_t &n)
+{
+	assertm(n <= static_cast<size_t>(std::numeric_limits<int>::max()),
+	        "SDL_net can't handle more bytes at a time.");
+	assert(data);
+	if (SDLNet_CheckSockets(listensocketset, 0)) {
+		const int result = SDLNet_TCP_Recv(mysock, data, static_cast<int>(n));
+		if(result < 1) {
+			isopen = false;
+			n = 0;
 			return false;
 		} else {
-			*size = static_cast<Bitu>(retval);
+			n = result;
 			return true;
 		}
-	}
-	else {
-		*size=0;
+	} else {
+		n = 0;
```

Comment:
Now takes in a reference to a `size_t` instead of `Bitu` pointer.
Applies your suggestion of checking the size before down-grading to an `int` for SDL.